pkg/controller/master/upgrade/upgrade_repo.go

```diff
@@ -335,8 +351,24 @@ func (r *Repo) deleteVM() error {
 		}
 	}
 
-	logrus.Infof("Delete upgrade repo VM %s/%s", vm.Namespace, vm.Name)
-	return r.h.vmClient.Delete(vm.Namespace, vm.Name, &metav1.DeleteOptions{})
+	err = r.h.vmClient.Delete(vm.Namespace, vm.Name, &metav1.DeleteOptions{})
+	if err == nil {
+		return nil
+	}
+	if apierrors.IsNotFound(err) {
```

Comment:
nit: beyond this change, technically, we can just delete the repo vm w/o checking if it exists first and rely on the error handling here.